configure.ac

```diff
@@ -17,16 +17,16 @@ AC_CANONICAL_HOST
 
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
-
 # Checks for libraries.
 PKG_CHECK_MODULES(OPENSSL, [libcrypto libssl])
 AC_CHECK_LIB([pthread], [pthread_create], [], [AC_MSG_ERROR([Cannot find libpthread.])])
 AC_CHECK_LIB([util], [forkpty], [], [AC_MSG_ERROR([Cannot find libutil.])])
 
 # Checks for header files.
-AC_CHECK_HEADERS([arpa/inet.h assert.h ctype.h errno.h fcntl.h getopt.h ifaddrs.h libutil.h limits.h mach/mach.h netdb.h net/if.h net/if_var.h netinet/in.h netinet/tcp.h net/route.h pty.h semaphore.h signal.h stdarg.h stddef.h stdint.h stdio.h stdlib.h string.h strings.h sys/ioctl.h sys/mutex.h syslog.h sys/socket.h sys/stat.h sys/time.h sys/types.h sys/wait.h unistd.h util.h])
-# fail without this one:
-AC_CHECK_HEADER([termios.h], [], AC_MSG_ERROR([Required header not present]))
+AC_CHECK_HEADERS([arpa/inet.h assert.h ctype.h errno.h fcntl.h getopt.h ifaddrs.h limits.h netdb.h net/if.h netinet/in.h netinet/tcp.h net/route.h pthread.h signal.h stdarg.h stdint.h stdio.h stdlib.h string.h sys/ioctl.h syslog.h sys/socket.h sys/stat.h sys/types.h sys/wait.h termios.h unistd.h], [], AC_MSG_ERROR([Required header not found]))
```

Comment:
I would prefer the complete list of headers. It's easier to check with a `grep -r '#include <' | awk -F\: '{print $2}' | sort -u`. They should appear either in the required ones or in the optional ones... BTW. `systemd/sd-daemon.h` should be added to the optional headers, then.

Another approach would be to reduce those lists and keep only the ones for which we actually have an `#if` statement, but I prefer having the HAVE_HEADER_H defined so that one can easily add the `#if` statement once it turns out to be necessary on some platform.

I must admit that those long lines in configure.ac are very difficult to review when changes happen there. Maybe we should write just one item on a line and mask the newlines with a backslash? Then it's easier to spot a single header that's inserted or removed, because it's always a line on its own.

Reply:
I tend to avoid at least the headers that are part of the C standard, but I'm OK with your suggestion. Perhaps an item per line would be easier to maintain, but I have no strong opinion, do as you wish.

Reply:
I have taken the C standard headers out of the check and put each one on a separate line.